lib/internal/modules/cjs/helpers.js

```diff
@@ -184,9 +192,33 @@ function normalizeReferrerURL(referrer) {
   return new URL(referrer).href;
 }
 
+// For error messages only - used to check if ESM syntax is in use.
+function hasEsmSyntax(code) {
+  const parser = acorn.Parser.extend(
+    privateMethods,
+    classFields,
+    staticClassFeatures
+  );
+  let root;
+  try {
+    root = parser.parse(code, { sourceType: 'module', ecmaVersion: 'latest' });
+  } catch {
+    return false;
+  }
+
+  for (const stmt of root.body) {
```

Comment:
Array.prototype[Symbol.iterator] can be hijacked; if root.body is an array, using `ArrayPrototypeSome` here would be preferable.